iceoryx_utils/test/integrationtests/test_resizeable_lockfree_queue_stresstest.cpp

```diff
@@ -0,0 +1,745 @@
+
+    bool testResult = true;
+    for (size_t i = 0; i < capacity; ++i)
+    {
+        // we expect each data item exactly numThreads + 1 times,
+        // the extra one is for the initially full queue
+        if (count[i] != numThreads + 1)
+        {
+            testResult = false;
+            break;
+        }
+    }
+
+    EXPECT_EQ(testResult, true);
+}
+
+/// @brief As the test before, but with an additional thread that periodically changes the capacity
+/// again it is checked that nothing is lost or created by accident.
+/// @note the tests are getting quite complicated but the complex setup is unavoidable
+/// in order to test the general case under load.
+TYPED_TEST(ResizeableLockFreeQueueStressTest, hybridMultiProducerMultiConsumer0verflowWithCapacityChange)
+{
+    using Queue = typename TestFixture::Queue;
+
+    auto& q = this->sut;
+    const int numThreads = 32;
+    const double popProbability = 0.45; // tends to overflow
+    const auto capacity = q.capacity();
+
+    std::atomic<bool> run{true};
+
+    std::vector<std::thread> threads;
+    std::vector<int> overflowCount(numThreads);
+    std::vector<std::list<Data>> itemVec(numThreads + 1);
+
+    // define capacities to cycle between
+    std::vector<uint64_t> capacities;
+    auto maxCapacity = q.maxCapacity();
+    for (uint64_t c = 1; c < maxCapacity; c *= 2)
+    {
+        capacities.push_back(c);
+    }
+    capacities.push_back(maxCapacity);
+
+    // fill the queue
+    Data d;
+    for (size_t i = 0; i < capacity; ++i)
+    {
+        d.count = i;
+        while (!q.tryPush(d))
+        {
+        }
+    }
+
+    for (int id = 1; id <= numThreads; ++id)
+    {
+        threads.emplace_back(randomWork<Queue>,
+                             std::ref(q),
+                             id,
+                             std::ref(run),
+                             capacity,
+                             std::ref(overflowCount[id - 1]),
+                             std::ref(itemVec[id - 1]),
+                             popProbability);
+    }
+
+    int id = numThreads + 1;
+    uint64_t numChanges;
+    threads.emplace_back(changeCapacity<Queue>,
+                         std::ref(q),
+                         std::ref(run),
+                         std::ref(itemVec[id - 1]),
+                         std::ref(capacities),
+                         std::ref(numChanges));
+
+    std::this_thread::sleep_for(std::chrono::seconds(this->runtime));
+
+    run = false;
+
+    for (auto& thread : threads)
+    {
+        thread.join();
+    }
+
+    // check whether all elements are there, but there is no specific ordering we can expect
+    // items are either in the local lists or the queue, in total we expect each count numThreads times
+
+    std::vector<int> count(capacity, 0);
+    auto popped = q.pop();
+    while (popped.has_value())
+    {
+        count[popped.value().count]++;
+        popped = q.pop();
+    }
+
+    // check the local lists
+    id = 1;
```

Comment:
`id` is set here, but not used. Is this a mistake?

Reply:
Leftover from were the test had output for specific threads, will remove.